hydra/utils.py

```diff
@@ -1,36 +1,34 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
 import logging.config
+import collections
 import os
 import sys
 from enum import Enum
 from pathlib import Path
 from typing import Any, Callable
 
 from omegaconf import OmegaConf
-from omegaconf._utils import is_structured_config
 
 from hydra._internal.utils import (
-    _convert_container_targets_to_strings,
     _convert_target_to_string,
-    _get_kwargs,
     _get_target_type,
     _locate,
     _pop_convert_mode,
+    _is_recursive,
 )
 from hydra.core.hydra_config import HydraConfig
-from hydra.errors import HydraException, InstantiationException
-from hydra.types import TargetConf
 
 log = logging.getLogger(__name__)
 
 
-class ConvertMode(Enum):
+class ConvertMode(str, Enum):
     # Use DictConfig/ListConfig
-    NONE = 0
+    NONE = "none"
     # Convert the OmegaConf config to primitive container, Structured Configs are preserved
-    PARTIAL = 1
+    PARTIAL = "partial"
     # Fully convert the OmegaConf config to primitive containers (dict, list and primitives).
-    ALL = 2
+    ALL = "all"
```

Comment:
curious about this one. can you tell me why you changed this?

Reply:
ah yes, just a simple trick I discovered recently to simplify the use of Enum (see https://docs.python.org/3/library/enum.html#others).

What you get from doing this is
1. equality : `ConvertMode.NONE == "none"` is now true
2. resolution `ConvertMode("none")` returns `ConvertMode.NONE`

Reply:
Interesting. OmegaConf is handling some of it for configs. I was not aware of this trick.

```
from dataclasses import *
from enum import Enum
from omegaconf import *

class Color(Enum):
    red = 1
    green = 2
    blue = 3

class StrColor(str, Enum):
    red = "red"
    green = "green"
    blue = "blue"


@dataclass
class Foo:
    color: Color = Color.red
    scolor: StrColor = StrColor.red

cfg : Foo = OmegaConf.structured(Foo)
print("cfg", cfg)

cfg.color = Color.blue
print("cfg", cfg)

# automatic conversion works
cfg.color = "green" # mypy complains
print("cfg", cfg)

# comparission does not work
assert cfg.color != "green"

## str color
print("str color")
cfg.scolor = "green" # mypy still complains
print("cfg", cfg)

# equality with strings works
assert cfg.scolor == "green"
```